internal/service/ch/ch.go

```diff
@@ -149,3 +149,28 @@ func (s *Service) getAggSignals(ctx context.Context, stmt string, args []any) ([
 	}
 	return signals, nil
 }
+
+// GetDeviceActivity returns the last 3 hour bucket during which device activity was recorded.
+func (s *Service) GetDeviceActivity(ctx context.Context, vehicleTokenId int, adManuf string) ([]*model.DeviceActivity, error) {
+	stmt, args := getDeviceActivityQuery(vehicleTokenId, adManuf)
+	rows, err := s.conn.Query(ctx, stmt, args...)
```

Comment:
[`QueryRow`](https://pkg.go.dev/github.com/ClickHouse/clickhouse-go/v2@v2.26.0/lib/driver#Conn) might be better.